src/UI/Panels/ClipboardPanel.cs

```diff
@@ -19,25 +20,76 @@ public class ClipboardPanel : UEPanel
         public override bool ShouldSaveActiveState => true;
         public override bool ShowByDefault => true;
 
-        private static Text CurrentPasteLabel;
+        private static List<GameObject> clipboardRows = new List<GameObject>();
 
         public ClipboardPanel(UIBase owner) : base(owner)
         {
         }
 
+        public static void SelectClipboardItem(int index)
+        {
+            selectedItem = index;
+        }
+
+        /// <summary>
+        /// Adds a new entry and selects it
+        /// </summary>
+        public static void AddNewEntry()
+        {
+            ClipboardItems.Add(null);
+            SelectClipboardItem(ClipboardItems.Count - 1);
+            UpdateCurrentPasteInfo();
+        }
         public static void Copy(object obj)
         {
-            Current = obj;
-            Notification.ShowMessage("Copied!");
+            if (obj == null || ClipboardItems.Count <= selectedItem)
+            {
+                ExplorerCore.Log($"The selected item index is outside the range! {ClipboardItems.Count} is the count.");
+                return;
+            }
+            Copy(obj,selectedItem, true);
+        }
+
+        public static void Copy(object obj, int index, bool showNotification)
+        {
+            if (obj == null)
+            {
+                return;
+            }
+            if (ClipboardItems.Count() <= selectedItem)
+            {
+                ClipboardItems.Insert(index,obj);
+            }
+            ClipboardItems[index] = obj;
+            if(showNotification)
+                Notification.ShowMessage("Copied to clipboard at index " + selectedItem + "!");
+            UpdateCurrentPasteInfo();
+        }
+
+        public static void CopyToEnd(object obj)
```

Comment:
I think this function is not used and should be removed.

Reply:
I made the CopyToEnd() method for when using the console, just because it makes it easier to add a new item and fill it at the end of the clipboard list. It could probably be removed, I just think it's useful for quick usage of the console.